common/Unit.cpp

```diff
@@ -237,6 +280,11 @@ void UnitBase::returnValue(int &retValue)
     if (TimeoutThread.joinable())
         TimeoutThread.join();
 
+    if (Global == GlobalKit)
+        GlobalKit = nullptr;
+    if (Global == GlobalWSD)
+        GlobalWSD = nullptr;
```

Comment:
If we have two instances in the same process, surely we want to delete both. The `delete Global;` below will only delete one.